Server.py

```diff
@@ -0,0 +1,226 @@
+import errno
+import select
+import socket
+import traceback
+
+import base
+import Client
+import CustomExceptions
+import util
+
+CLOSE, SERVER, CLIENT = range(3)
+CRLF = "\r\n"
+END_HEARDER = 2 * CRLF
+
+
+class Server(base.Base):
+    _database = {}
+
+    def __init__(
+        self,
+        buff_size,
+        base_directory
+    ):
+        super(Server, self).__init__()
+        self._buff_size = buff_size
+        self._run = True
+        self.logger.info("Initialized server, buff size '%d'" % buff_size)
+        self._base_directory = base_directory
+
+    def terminate(self):
+        self._run = False
+
+    def add_server(
+        self,
+        our_address=("localhost", 80)
+    ):
+        """ Creat server side socket and add it to the database. \n
+            our_address: tupple of the address that the server will do bind to.
+                         format (address, port), default (localhost, 80)
+        """
+        s = util.creat_nonblocking_socket()
+        s.bind(our_address)
+        s.listen(1)
+        self._add_to_database(s)
+        self.logger.info("Created server on address '%s:%s'" % our_address)
+
+    def _add_to_database(self, s, state=SERVER, peer=None):
+        """
+                Add the socket to the database, structre will be:
+                {socket :  {
+                "buff":buffer will contain the data we need to send,
+                "state": state of the socket, CLOSE, SERVER or CLIENT,
+                "file": file_object to send,
+                "peer": CLIENT, the server we are connected to,
+                        SERVER, list of the sockets that connected to server
+                        CLOSE, None
+                }
+        """
+        if state not in (CLOSE, SERVER, CLIENT):
+            raise RuntimeError("State not found")
+        self._database[s] = {
+            "state": state,
+        }
+        entry = self._database[s]
+        if state == SERVER:
+            entry["peer"] = []
+        elif state == CLIENT:
+            entry["client"] = Client.Client(s, self._buff_size)
```

Comment:
Yes it does, diffrent states act diffrently.
We don't want to get information from a socket in CLOSE state, only send.
In SERVER (or active) socket we don't want to send, only recive. 
And with with CLIENT we want to know if we can send or recive depends on the state